File: programmers/js/level00/120880.js
// 특이한 정렬
function solution(numlist, n) {
  return numlist
    .map(num => [num, Math.abs(num - n)])
    .sort((a, b) => a[1] - b[1] || b[0] - a[0])
    .map(([num, order]) => num);
}

// 다른 풀이
// function solution(numlist, n) {
//   return numlist.sort((a, b) => Math.abs(a - n) - Math.abs(b - n) || b - a);
// }

File: programmers/java/level00/120880.java
// 특이한 정렬

import java.util.Arrays;

class Solution {
    public int[] solution(int[] numlist, int n) {
        return Arrays.stream(numlist)
            .boxed()
            .map(num -> new int[]{num, Math.abs(n - num)})
            .sorted((a, b) -> a[1] - b[1] == 0 ? b[0] - a[0] : a[1] - b[1])
            .mapToInt(arr -> arr[0])
            .toArray();
    }
}